src/vdaf.rs

```diff
@@ -193,27 +203,52 @@ pub enum PrepareTransition<S, M, O> {
 }
 
 /// An aggregate share that can merged with aggregate shares of the same type.
-pub trait Aggregatable {
+pub trait Aggregatable: Clone + Debug + Serialize + DeserializeOwned {
+    /// Type of VDAF output shares that can be accumulated into an aggregatable.
+    type OutputShare;
+
     /// Update an aggregate share by merging it with another (`agg_share`).
     fn merge(&mut self, agg_share: &Self) -> Result<(), VdafError>;
+
+    /// Update an aggregate share by merging it with an output share
+    fn accumulate(&mut self, output_share: &Self::OutputShare) -> Result<(), VdafError>;
 }
 
-impl<F: FieldElement> Aggregatable for Vec<F> {
-    fn merge(&mut self, agg_share: &Vec<F>) -> Result<(), VdafError> {
-        if self.len() != agg_share.len() {
+/// An output share suitable for VDAFs whose output shares are a vector of field
```

Comment:
`F` doesn't have the `FieldElement` trait bound, so in theory this type is more general. I remember you saying that implementations should have the bound and not the type, so I'm not asking for a change to the type. However, it may be more valuable to make the comment more general, and maybe name the type accordingly.

What about

```rust
/// An output share comprised of a vector of `F` elements.
type OutputShareVec<F>(pub vec<F>);
```

Similarly for `AggregateShare` below.

Reply:
This could be a case where it's appropriate to put the trait bound in the struct, since it informs what the struct contains and not just its behavior. Being explicit about `FieldElement` here would make this more readable.

Reply:
I spoke too soon! This is exactly the case where we don't want to specify `F: FieldElement` because then we can't derive `Deserialize`. I'll amend the comment as you suggested.